src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -45,4 +46,11 @@
      * @throws IOException
      */
     Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException;
+
+    /**
+     * Validate the query.
+     * @return {@code null} if query is valid, otherwise {@link ActionRequestValidationException} containing error messages,
+     * e.g. if fields that are needed to create the lucene query are missing.
+     */
+    ActionRequestValidationException validate();
```

Comment:
sorry for changing my mind, what do you think about having our own exception here e.g. `QueryValidationException` ? when we'll hook into the search request validation we can just wrap it in the ActionRequestValidationException or something along those lines. I wouldn't query builders to depend on transport action packages ideally.